core/primitives/src/ed25519.rs

```diff
@@ -41,7 +41,7 @@ use crate::{crypto::{Public as TraitPublic, UncheckedFrom, CryptoType, Derive}};
 type Seed = [u8; 32];
 
 /// A public key.
-#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Encode, Decode, Default)]
+#[derive(Eq, PartialOrd, Ord, Clone, Encode, Decode, Default)]
```

Comment:
A better fix is to use `cfg_attr` to guard `derive(Hash)` here, instead of manually implementing `Hash` or `PartialEq`.